### ISSUE TITLE
CARMA is incompatible with RRTMGP

### ISSUE THREAD

**Issue by brhillman** [E3SM-Project/E3SM#3533]
CARMA is expected to fail when configured with RRTMGP. This is because `carma_register` attempts to grab information about the spectral bands in use from the `radconstants` module, but with RRTMGP this information is not setup until the inputdata is read by `radiation_init` which comes after the register routines. Thus, calling `carma_register` will attempt to grab uninitialized data from `radconstants`. There a few possible fixes for this: 1) move the file i/o for RRTMGP to the `radiation_register` routine, and move `radiation_register` before `carma_register`; 2) revert to hard-coding the band information into `radconstants`. I'm not a huge fan of hard-coding these values since it makes the code less flexible, but we are probably unlikely to change these values, and perhaps we could then add a check at initialization to make sure the hard-coded values match what is read from the inputfiles.

**Comment by PeterCaldwell** [comment]
I've been wondering for years - does anyone actually use CARMA? Can we strip it out? Just curious.

**Comment by brhillman** [comment]
@whannah1 and I were talking about this the other day. It does not seem to be used anywhere, and I could not find any existing compsets or tests that configure it.

**Comment by tangq** [comment]
Toon's team uses CARMA with CESM. See this: https://science.sciencemag.org/content/sci/suppl/2019/08/07/365.6453.587.DC1/aax1748_Yu_SM.pdf

Not sure if it's publicly available or not.

**Comment by brhillman** [comment]
Actually it looks like we do have some compsets in E3SM that theoretically use it, not sure if these are ever used though:
```
bhillma@cori01:~/codes/e3sm/branches/master/components/cam/cime_config> grep carma -ri .
./config_component.xml:      <value compset="_CAM[45]%WCBC"  >-carma bc_strat</value>
./config_component.xml:      <value compset="_CAM4%WCSF"     >-chem waccm_mozart_sulfur -carma sulfate</value>
./config_component.xml:      <value compset="2013_CAM4%WCBC"   >waccm_carma_bc_2013_cam4</value>
./config_component.xml:    <desc compset="_CAM[45]%WCBC">CAM WACCM with the stratospheric black carbon CARMA model:</desc>
./config_component.xml:    <desc compset="_CAM[45]%WCSF">CAM WACCM with sulfur chemistry and the sulfate CARMA model:</desc>
```

**Comment by PeterCaldwell** [comment]
Thanks Qi. Yeah, I know CARMA is Brian Toon's thing and that it was used in CESM back when dinosaurs roamed the earth. I'm just wondering whether anyone in *E3SM* uses it or plans to ever use it. I suspect not. I bet it was just inherited from the CESM split.

**Comment by PeterCaldwell** [comment]
@brhillman - does WACCM even work in E3SM? A git blame on those compsets might be interesting.

**Comment by tangq** [comment]
The paper above was published in 2019. They must be using it now. Likely not in the standard CESM version though.

**Comment by brhillman** [comment]
@PeterCaldwell I don't think so, at least from informal conversations. Git blame on the lines that mention carma:
```
aacdbd9f98 (Robert Jacob 2016-07-06 20:54:05 -0500  96)       <value compset="_CAM[45]%WCBC"  >-carma bc_strat</value>
aacdbd9f98 (Robert Jacob 2016-07-06 20:54:05 -0500  97)       <value compset="_CAM4%WCSF"     >-chem waccm_mozart_sulfur -carma sulfate</value>
aacdbd9f98 (Robert Jacob 2016-07-06 20:54:05 -0500 234)       <value compset="2013_CAM4%WCBC"   >waccm_carma_bc_2013_cam4</value>
aacdbd9f98 (Robert Jacob 2016-07-06 20:54:05 -0500 334)     <desc compset="_CAM[45]%WCBC">CAM WACCM with the stratospheric black carbon CARMA model:</desc>
aacdbd9f98 (Robert Jacob 2016-07-06 20:54:05 -0500 335)     <desc compset="_CAM[45]%WCSF">CAM WACCM with sulfur chemistry and the sulfate CARMA model:</desc>
```
I assume we just inherited all of these compsets and many of them are probably broken...

**Comment by PeterCaldwell** [comment]
Yeah, I think we should delete it but it would probably be good to get an opinion from someone at PNNL who is more connected to aerosol stuff. @hwangacme ?

**Comment by rljacob** [comment]
We still have CAM  compsets and component configs from when we started E3SM that were never cleaned up.

**Comment by rljacob** [comment]
@yfenganl any opinion on above?

**Comment by hwangacme** [comment]
@PeterCaldwell @rljacob 
This aerosol option (CARMA) has never been tested in E3SM. As far as I know, there is no plan to use it for ongoing or future developments, so it makes sense to remove it.

**Comment by eroesler** [comment]
If CARMA is taken out,  are sulfate aerosols and chemistry treated in the stratosphere in E3SM through another module?-- Linoz?

**Comment by hwangacme** [comment]
@eroesler Linoz is used in E3SMv1. More comprehensive chemistry is being developed for E3SMv3. I don't think the chemistry is bundled with CARMA.
